### PR TITLE
Avoid returning type symbols for js configured projects

### LINKED ISSUE #10606: TSServer: for JS projects workspace symbol route shouldn't contain types from any d.ts files
**TypeScript Version:**  1.8.10

- create a JS project (jsconfig.json)
- add a js file
- open in tsserver and request workspace symbols
- the returned symbols contain TS types that you can't use in JS (e.g. have no static members or can't be newed). Showing them doesn't make much sense in a JS project.

### PR DESCRIPTION
**Pull request by zhengbli**
Fixes #10606

### CODE REVIEW

File: src/harness/unittests/tsserverProjectSystem.ts [hunk 1]
```diff
@@ -1699,7 +1706,36 @@ namespace ts.projectSystem {
                 "File '/a/cache/node_modules/@types/lib/index.tsx' does not exist.",
                 "File '/a/cache/node_modules/@types/lib/index.d.ts' exist - use it as a name resolution result.",
             ]);
-            checkProjectActualFiles(proj, [ file1.path, lib.path ]);
+            checkProjectActualFiles(proj, [file1.path, lib.path]);
+        });
+    });
+
+    describe("nagivateTo for javascript project", () => {
```
Comment (vladima) [inline src/harness/unittests/tsserverProjectSystem.ts]: `navigate`

File: src/server/project.ts [hunk 2]
```diff
@@ -134,6 +134,13 @@ namespace ts.server {
         abstract getProjectName(): string;
         abstract getTypingOptions(): TypingOptions;
 
+        getNavigateToItems(searchValue: string, maxResultCount: number) {
```
Comment (vladima) [inline src/server/project.ts]: I would not put this method in project but rather expose:
```ts
isJsOnlyProject() {  
    return allRootFilesAreJsOrDts(this); // probably can be cached and invalidated when project structure changes
}
```

File: src/server/project.ts [hunk 3]
```diff
@@ -543,6 +550,9 @@ namespace ts.server {
         private projectFileWatcher: FileWatcher;
         private directoryWatcher: FileWatcher;
         private directoriesWatchedForWildcards: Map<FileWatcher>;
+
+        public readonly isJSProject: boolean;
```
Comment (vladima) [inline src/server/project.ts]: is it used?

File: src/harness/unittests/tsserverProjectSystem.ts [hunk 4]
```diff
@@ -1699,7 +1706,36 @@ namespace ts.projectSystem {
                 "File '/a/cache/node_modules/@types/lib/index.tsx' does not exist.",
                 "File '/a/cache/node_modules/@types/lib/index.d.ts' exist - use it as a name resolution result.",
             ]);
-            checkProjectActualFiles(proj, [ file1.path, lib.path ]);
+            checkProjectActualFiles(proj, [file1.path, lib.path]);
+        });
+    });
+
+    describe("nagivate for javascript project", () => {
```
Comment (vladima) [inline src/harness/unittests/tsserverProjectSystem.ts]: `navigate`

Comment (zhengbli) [inline src/harness/unittests/tsserverProjectSystem.ts]: Aha I misunderstood.
Time to find a spell check plugin ...

File: src/server/project.ts [hunk 5]
```diff
@@ -288,6 +293,7 @@ namespace ts.server {
                 hasChanges = this.updateGraphWorker() || hasChanges;
             }
             if (hasChanges) {
+                this._isJsOnlyProject = allRootFilesAreJsOrDts(this);
```
Comment (vladima) [inline src/server/project.ts]: actually after thinking about it a bit more better places to invalidate the value will be `addRoot` and `removeRoot` - there we can set `this._isJsOnlyProject` to `undefined` and compute the value on demand when it is requested

**Comment by vladima** [comment]
:+1:

**Pull request merged**